File: 06-subprogramas/subProgramas.py

def modificar_imutavel(numero):
    print(f"  -> Dentro da função (antes): {numero}")
    numero = 100  
    print(f"  -> Dentro da função (depois): {numero}")

def modificar_mutavel(lista):
    print(f"  -> Dentro da função (antes): {lista}")
    lista.append(30) 
    print(f"  -> Dentro da função (depois): {lista}")

valor_int = 10
print(f"Valor int (antes): {valor_int}")
modificar_imutavel(valor_int)
print(f"Valor int (depois): {valor_int}")

print("-" * 20)

valor_list = [10, 20]
print(f"Valor list (antes): {valor_list}")
modificar_mutavel(valor_list)
print(f"Valor list (depois): {valor_list}")

File: 06-subprogramas/subProgramas.cs
using System;

public class GerenciadorDeValores
{
    public void ModificarValor(int numero)
    {
        numero = 100;
    }

    public void ModificarComRef(ref int numero)
    {
        numero = 100;
    }
}

public class Teste
{
    public static void Main(string[] args)
    {
        var gerenciador = new GerenciadorDeValores();

        int val = 10;
        Console.WriteLine($"Valor original (antes): {val}");
        gerenciador.ModificarValor(val);
        Console.WriteLine($"Valor original (depois de 'ModificarValor'): {val}");

        Console.WriteLine(new string('-', 20));

        int refVal = 10;
        Console.WriteLine($"Valor ref (antes): {refVal}");
        gerenciador.ModificarComRef(ref refVal);
        Console.WriteLine($"Valor ref (depois de 'ModificarComRef'): {refVal}");
    }
}
